solidity-v1/dashboard/src/pages/applications/ThresholdApplicationPage.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useMemo, useState } from "react"
+import { useSelector } from "react-redux"
+import EmptyStatePage from "./EmptyStatePage"
+import { isSameEthAddress } from "../../utils/general.utils"
+import { LoadingOverlay } from "../../components/Loadable"
+import DataTableSkeleton from "../../components/skeletons/DataTableSkeleton"
+import AuthorizeThresholdContracts from "../../components/threshold/AuthorizeThresholdContracts"
+import ThresholdAuthorizationHistory from "../../components/threshold/ThresholdStakingAuthorizationHistory"
+import { MODAL_TYPES } from "../../constants/constants"
+import { useModal } from "../../hooks/useModal"
+import AuthorizeStakesBanner from "../../components/threshold/AuthorizeStakesBanner"
+
+const ThresholdApplicationPage = () => {
+  const [selectedOperator, setOperator] = useState({})
+  const { openModal } = useModal()
+  const thresholdAuthState = useSelector(
+    (state) => state.thresholdAuthorization
+  )
+
+  const authorizeContract = useCallback(
+    async (data, awaitingPromise) => {
+      const {
+        operatorAddress,
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.AuthorizeAndStakeOnThreshold, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: false,
+      })
+    },
+    [openModal]
+  )
+
+  const stakeToT = useCallback(
+    async (data, awaitingPromise) => {
+      const {
+        operatorAddress,
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.StakeOnThresholdWithoutAuthorization, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: true,
+      })
+    },
+    [openModal]
+  )
+
+  const thresholdAuthData = useMemo(() => {
+    const thresholdData = thresholdAuthState.authData.filter((dataObj) => {
+      return !dataObj.isStakedToT || !dataObj.contracts[0].isAuthorized
```

Comment:
Do we need `!dataObj.isStakedToT || !dataObj.contracts[0].isAuthorized`? Isn't `dataObj.contracts[0].isAuthorized` enough?`

Reply:
I tWe display both - stakes that are unauthorized and not moved to t, and stakes that are authorized but not moved to t.

I think you meant to just use `!dataObj.contracts[0].isStakedToT` instead of `dataObj.contracts[0].isAuthorized`, right? I think in this case it would work, because we basically shows are stakes that are not moved to T, doesn't matter the authorization status.